csunplugged/topics/views.py

```diff
@@ -337,3 +340,36 @@ class OtherResourcesView(generic.DetailView):
     model = Topic
     template_name = 'topics/topic-other-resources.html'
     slug_url_kwarg = 'topic_slug'
+
+
+def glossary(request, **kwargs):
+    """Provide glossary view.
+
+    If the request includes the query parameter 'term', then
+    a JSON response is sent containing data for the requested term.
+    Otherwise the glossary page of all terms is rendered and sent.
+
+    Returns:
+        If 'term' parameter: JSON response of glossary term data.
+        Otherwise HTTP response of glossary page of all terms.
+    """
+    # If term parameter, then return JSON
+    if "term" in request.GET:
+        glossary_slug = request.GET["term"]
+        glossary_item = get_object_or_404(
+            GlossaryTerm,
+            slug=glossary_slug
+        )
+        data = {
+            "slug": glossary_slug,
+            "term": glossary_item.term,
+            "definition": render_html_with_static(glossary_item.definition)
+        }
+        return JsonResponse(data)
+    # Otherwise return glossary webpage with all terms
```

Comment:
so...if I request a term that doesn't exist, then I'm taken to the glossary page (when I'm expecting just a modal to pop up) ? Is that correct? That seems like odd behaviour.